Rewards flow

oetyng opened the pull request:
Changes to enable the use of multisig transfer actor.

oetyng committed:
refactor: split into two queries
- These need to be handled differently

oetyng committed:
chore: remove messaging mod

oetyng committed:
remove deprecated ElderDuty variant

oetyng committed:
feat(multi_sig): add signed transfer shares

oetyng committed:
feat(transfers): add as_share api to debit+credit

Yoga07 commented on src/keys/keypair.rs:
```diff
@@ -27,8 +27,6 @@ use threshold_crypto::{self, serde_impl::SerdeSecret};
     /// Ed25519 keypair.
     Ed25519(ed25519_dalek::Keypair),
-    /// BLS keypair.
-    Bls(BlsKeypair),
```
Not sure why we are removing this?
(On a side note, we use BLS keypairs a lot at sn_client(esp for testing), removing this would induce a big refactor at client I think)

joshuef commented on src/transfer.rs:
```diff
@@ -25,6 +25,15 @@ pub type CreditId = [u8; 256 / 8];
 /// Msg, containing any data to the recipient.
 pub type Msg = String;
 
+///
```
missing docs here

oetyng committed:
chore: add + cleanup comments

oetyng committed:
refactor(transfer): do not use replicaevents
- Instead of synching subjective views (replicas' events), we synch consolidated data - i.e. the agreed credits and debits, signed by the replicas.
